src/backend/access/transam/xlogutils.c

```
@@ -96,7 +99,8 @@ log_invalid_page(RelFileNode node, ForkNumber forkno, BlockNumber blkno,
 	if (reachedConsistency)
 	{
 		report_invalid_page(WARNING, node, forkno, blkno, present);
-		elog(PANIC, "WAL contains references to invalid pages");
+		elog(ignore_invalid_pages ? WARNING : PANIC,
```

given upstream didn't implement so maybe tomorrow problem - but isn't hash-table ever growing in size with this ignored error in recovery mode. Shouldn't we be clearing the entries from it if don't care about there presence.

---

Yes we can don't save log record, but it lead to either
1) logging of reference to invalid page that actually doesn't (false positive);
2) or loosing ability to determining invalid reference at all (no logging)

second option looks ok to use-case described in this pr, because it save 3Mb per 1Gb of pages in original whole database in a very very worst case (i calc 3mb from 26 byte per key per 8Kb page).